### PR TITLE
fix: windows build

### PR DESCRIPTION
**Pull request by estrada9166**
- Closes  [UNIFY-1377](https://cypress-io.atlassian.net/jira/software/projects/UNIFY/boards/20?selectedIssue=UNIFY-1377)

### Additional details

Currently, the build on windows machines is not working as expected because there's a bash script that is not supported unless extra packages are installed on the machine.

Also, `rm` is not supported at all, so there's the need to update the usage of it for `rimraf` which is fully supported!

Update failing migration test where the file path is not matching on windows machines

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by elevatebart** [comment]
I believe bash was already required for `run-if-not-ci.sh`

**Comment by elevatebart** [comment]
I like TypeScript as it is easier to debug than bash.

File: packages/icons/scripts/build.ts
Comment (ZachJW34) [inline packages/icons/scripts/build.ts:17]: Not related to this PR, but does this platform specific build have any effect on what the binary expects? When we build the Cypress binary for windows, will it have `dist/icons/cypress.icns` available and is it needed? Maybe @elevatebart has an idea

**Review by elevatebart — DISMISSED** [review]

File: package.json
Comment (tbiethman) [inline package.json:40]: The 'clean-deps' command here at root is still using `rm`

File: packages/config/package.json
Comment (tbiethman) [inline packages/config/package.json:12]: These clean commands that delete implementation files are wild; are they actively used?

File: package.json
Comment (estrada9166) [reply package.json:40]: Yeah! I didn't changed bc it was changed 2 years ago and seems not to be an issue - but may worth it changing to be consistent across all the packages

image

File: packages/config/package.json
Comment (estrada9166) [reply packages/config/package.json]: It seems those are actively used - the screenshot is running `yarn` in `10.0-release` branch

image

File: packages/icons/scripts/build.ts
Comment (elevatebart) [reply packages/icons/scripts/build.ts:17]: @ZachJW34, the answer is no and it does not need it. the only moment where we "need" the icns file is when we package the electron app for mac. `icns` is the mac specific icon format that contains multiple definitions. Since this electron app can only be built and signed on Mac, I am not afraid of this happening.

File: packages/config/package.json
Comment (elevatebart) [reply packages/config/package.json]: Yeah, we don't use dist folders at cypress to avoid navigating to type files when changing packages. source maps would be a solution, but it does not seem to be one we considered for dev. we prefer using ts-node.

Comment (elevatebart) [reply packages/config/package.json]: When we run build-prod, on local, it will create a bunch of files that need to be cleaned up.

**Review by elevatebart — APPROVED** [review]

**Review by elevatebart — APPROVED** [review]

**Review by lmiller1990 — APPROVED** [review]
Code seems fine but how did you test this on windows? Are you using windows? Should we enable windows for this PR using `circle.yml` to see if it actually all works?

**Comment by elevatebart** [comment]
I tested it on windows.

**Comment by lmiller1990** [comment]
Ok - so once this merges, and windows runs on the 10.0-release it should be green? Are there any outstanding windows problems?

I'll leave merging this to @estrada9166 :pray:

**Comment by marktnoonan** [comment]
I'm going to make a branch off of this now that runs the windows tests, so we can see them run in CI before merging to 10.0-release without modifying this branch's `circle.yml`, I will update with results when they have completed.

File: scripts/run-postInstall.js
Comment (marktnoonan) [inline scripts/run-postInstall.js:4]: Excluding CI for this caused the `postinstall` step in the `windows-node-modules-install` job to fail on Windows in CI due to the non-windows version of this code running. Can we remove the CI check safely, or do we need another condition for windows on CI?

Here is a copy [run based on a copy of this branch with windows tests enabled](https://app.circleci.com/pipelines/github/cypress-io/cypress/35384/workflows/5275720e-e097-4a07-a8af-860b05258000/jobs/1408820) in CI. We see this error:

```
Error: Command failed: patch-package && ./scripts/run-if-not-ci.sh yarn-deduplicate --strategy=highest && yarn clean && gulp postinstall && ./scripts/run-if-not-ci.sh yarn build
```

**Review by marktnoonan — CHANGES_REQUESTED** [review]
Looks like we need a little more logic in `scripts/run-postInstall.js` - or just to remove the CI check, I haven't looked in detail yet, just wanted to make sure we don't merge yet. I'll look into it a bit more and try some changes on the `win-fix-windows-build-ci` branch. If that starts passing I'll make a PR into this with the changes.

**Review by tbiethman — APPROVED** [review]

**Review by marktnoonan — APPROVED** [review]
Approving this now as the windows build succeeds, I expect a few windows tests may continue to fail in CI after this PR based on letting CI run already in a different branch, but that's fine, this PR is not intended to fix all outstanding windows test issues.

**Pull request merged**
